src/FakeItEasy/Configuration/ArgumentCollection.cs

```diff
@@ -78,14 +75,17 @@ public int Count
         }
 
         /// <summary>
-        ///   Returns an enumerator that iterates through the collection or arguments.
+        ///   Returns an enumerator that iterates through the collection of argument values.
         /// </summary>
         /// <returns>
         ///   A <see cref = "T:System.Collections.Generic.IEnumerator`1" /> that can be used to iterate through the collection.
         /// </returns>
         public IEnumerator<object> GetEnumerator()
         {
-            return this.arguments.Cast<object>().GetEnumerator();
+            return GetEnumerable(this.arguments).GetEnumerator();
+
+            // Like using Cast<T>(), but without expensive LINQ allocations or risky casts
+            IEnumerable<T> GetEnumerable<T>(T[] arr) => arr;

```

Comment:
Fancy, but isn't it the same as `((IEnumerable<object>)this.arguments).GetEnumerator();`?

Reply:
So it is. I'll amend.